File: 保持城市天际线.cpp
class Solution {
public:
    int maxIncreaseKeepingSkyline(vector<vector<int>>& grid) {
        int HeightR[50];
        int HeightC[50];
        int length = grid.size();
        memset(HeightC, 0, sizeof(HeightC));
        memset(HeightR, 0, sizeof(HeightR));
        for(int i = 0; i < length; i++){
            for(int j = 0; j < length; j++){
                HeightR[i] = max(HeightR[i], grid[i][j]);
                HeightC[j] = max(HeightC[j], grid[i][j]);
            }
        }
        int ans = 0;
        for(int i = 0; i < length; i++){
            for(int j = 0; j < length; j++){
                ans += min(HeightC[j], HeightR[i]) - grid[i][j];
            }
        }
        return ans;
    }
};

File: 保持城市天际线.py
class Solution:
    def maxIncreaseKeepingSkyline(self, grid: List[List[int]]) -> int:
        n = len(grid)
        max_row = [max(grid[i]) for i in range(n)]
        max_col = [max(grid[i][j] for i in range(n)) for j in range(n)]
        return sum([min(max_row[i//n]-grid[i//n][i%n], max_col[i%n]-grid[i//n][i%n]) for i in range(n**2)])
